cmd/otelcorecol/go.mod

```
@@ -2,9 +2,9 @@
 
 module go.opentelemetry.io/collector/cmd/otelcorecol
 
-go 1.22.0
+go 1.23
```

I don't think we want this changed in this PR?
```suggestion
go 1.22.0
```

---

Oh but it looks like backoff v5 requires Go 1.23

```console
$ make gotidy
...
go: github.com/cenkalti/backoff/v5@v5.0.0 requires go >= 1.23 (running go 1.22.10)
...
```

---

I know, that is why I will not merge this PR now.